include/RAJA/policy/hip/kernel/For.hpp

```diff
@@ -153,79 +92,6 @@ struct HipStatementExecutor<
 };
 
 
-/*
- * Executor for thread work sharing loop inside HipKernel.
- * Provides a block-stride loop (stride of blockDim.xyz) for
- * each thread in xyz.
- * Assigns the loop index to offset ArgumentId
- */
-template <typename Data,
-          camp::idx_t ArgumentId,
-          int ThreadDim,
-          typename... EnclosedStmts,
-          typename Types>
-struct HipStatementExecutor<
-    Data,
-    statement::For<ArgumentId, RAJA::hip_thread_xyz_loop<ThreadDim>, EnclosedStmts...>,
-    Types> {
-
-  using stmt_list_t = StatementList<EnclosedStmts...>;
-
-  // Set the argument type for this loop
-  using NewTypes = setSegmentTypeFromData<Types, ArgumentId, Data>;
-
-  using enclosed_stmts_t =
-      HipStatementListExecutor<Data, stmt_list_t, NewTypes>;
-
-  using diff_t = segment_diff_type<ArgumentId, Data>;
-
-
-  static
-  inline RAJA_DEVICE void exec(Data &data, bool thread_active)
-  {
-    // block stride loop
-    diff_t len = segment_length<ArgumentId>(data);
-    diff_t i_init = get_hip_dim<ThreadDim>(dim3(threadIdx.x,threadIdx.y,threadIdx.z));
-    diff_t i_stride = get_hip_dim<ThreadDim>(dim3(blockDim.x,blockDim.y,blockDim.z));
-
-    // Iterate through block stride of chunks
-    for (diff_t ii = 0; ii < len; ii += i_stride) {
-      diff_t i = ii + i_init;
-
-      // execute enclosed statements if any thread will
-      // but mask off threads without work
-      bool have_work = i < len;
-
-      // Assign the x thread to the argument
-      data.template assign_offset<ArgumentId>(i);
-
-      // execute enclosed statements
-      enclosed_stmts_t::exec(data, thread_active && have_work);
-    }
-  }
-
-
-  static
-  inline
-  LaunchDims calculateDimensions(Data const &data)
-  {
-    diff_t len = segment_length<ArgumentId>(data);
-
-    // request one thread per element in the segment
-    LaunchDims dims;
-    set_hip_dim<ThreadDim>(dims.threads, len);
-
-    // but, since we are looping, we only need 1 thread, or whatever
-    // the user specified for MinThreads
-    set_hip_dim<ThreadDim>(dims.min_threads, 1);
```

Comment:
In the new stuff I do not set the min to 1 and I'm not certain how to do that in general without replicating lots of code, any ideas? Here is where I set threads https://github.com/LLNL/RAJA/pull/1334/files#diff-d3afa21cfcf86a8ee767f4b7b1fd7b6a500faa6a490f9abc7e24ff778e0aa7c9R343 .